src/external-server/external-server.ts

```diff
@@ -57,6 +59,11 @@ export abstract class ExternalServer {
         protected stderr?: string
     ) {
         ExternalServer.cleanupFns.push(this.cleanup.bind(this));
+        this._isDockerDesktop = isDockerDesktop();
+    }
+
+    public get isDockerDesktop(): boolean {
```

Comment:
I didn't mean the getter to just return the value of a private property, but also set the value of it so that it doesn't have to be set elsewhere. And so that this getter is the only place responsible for checking.